mhkit/wave/graphics.py

```diff
@@ -123,3 +123,143 @@ def plot_matrix(M, xlabel='Te', ylabel='Hm0', zlabel=None, show_values=True,
+        assert isinstance(lambda_w, (np.ndarray, float, int, np.int64)), 'lambda_w must be a real numeric type'
+        assert isinstance(D, (np.ndarray, float, int, np.int64)), 'D must be a real numeric type'
+        assert isinstance(ax, (type(None), plt.Axes))
+        if isinstance(H, np.ndarray):
+            assert H.squeeze().shape == lambda_w.squeeze().shape \
+            and H.squeeze().shape == D.squeeze().shape, 'H, lambda_w, and D must be same shape'
+        else:
+            H = np.array([H])
+            lambda_w = np.array([lambda_w])
+            D = np.array([D])
+
+        mvals = pd.DataFrame([H.flatten(),lambda_w.flatten(),D.flatten()],
+                         index=['H','lambda_w','D']).transpose()
+
+    if ax is None:
+        plt.figure()
+        ax = plt.gca()
+
+    ax.set_xscale('log')
+    ax.set_yscale('log')
+    x = np.logspace(-2, 1, 1000)
+
+    # upper bound of low drag region    
+    ldv = 20
+    y_small_drag = ldv*np.ones_like(x)
+    ax.plot(x[x < 0.14 * np.pi / ldv], y_small_drag[x < 0.14 * np.pi / ldv],
+            'k--')
+    ax.text(0.0125, 30, 'drag', ha='center', va='top', fontstyle='italic',
+            fontsize='small')
+
+    # upper bound of small drag region
+    sdv = 1.5
+    y_small_drag = sdv*np.ones_like(x)
+    ax.plot(x[x < 0.14 * np.pi / sdv], y_small_drag[x < 0.14 * np.pi / sdv],
+            'k--')
+    ax.text(0.02, 7, 'inertia \n& drag', ha='center', va='top',
+            fontstyle='italic', fontsize='small')
+     
+    # upper bound of neglible drag region   
+    ndv = 0.25
+    y_small_drag = ndv*np.ones_like(x)
+    ax.plot(x[x < 0.14 * np.pi / ndv], y_small_drag[x < 0.14 * np.pi / ndv],
+            'k--')
+    ax.text(8e-2, 0.7, 'large\ninertia', ha='center', va='top',
+        fontstyle='italic', fontsize='small')
+
+    ax.text(8e-2, 6e-2, 'all\ninertia', ha='center', va='top',
+            fontstyle='italic', fontsize='small')
+
+    # left bound of diffraction region
+    drv = 0.5
+    y_diff_reg = np.array([1e-2, 0.14 * np.pi / drv])
+    x_diff_reg = 0.5 * np.ones_like(y_diff_reg)
+    ax.plot(x_diff_reg, y_diff_reg, 'k--')
+    ax.text(2, 6e-2, 'diffraction', ha='center', va='top', fontstyle='italic',
+            fontsize='small')
+
+    # deep water breaking limit (H/lambda_w = 0.14)
+    y_breaking = 0.14 * np.pi / x
+    ax.plot(x, y_breaking, 'k-')
+    ax.text(1, 7, 'wave\nbreaking\n$H/\lambda_w > 0.14$', ha='center', va='top',
+            fontstyle='italic', fontsize='small')
+
+    for index, row in mvals.iterrows():
+        xx = row['H']/row['D']
+        yy = np.pi*row['D']/row['lambda_w']
+        lab = '$H = %.2g,\\,lambda_w = %.2g,\\,D = %.2g$' % (row['H'], row['lambda_w'], row['D'])
+        ax.plot(xx, yy, 'o', label=lab)
+        # print(row['c1'], row['c2'])
+
+    if index > 0:
+        ax.legend(fontsize='xx-small', ncol=2)
+
+    ax.set_xlim((0.01, 10))
+    ax.set_ylim((0.01, 50))
```

Comment:
@ryancoe currently I can put in numbers that may be outside the range of this plot. A user may not notice unrealistic values in an array or DataFrame. 

Would the correct response be to send a warning that a value is outside of these hardcoded limits or to increase the limits of the plot?

Reply:
Example if I remove the hardcoded limits
![image](https://user-images.githubusercontent.com/13438942/81346789-246fbf80-9078-11ea-9173-5f251c4bcf73.png)

Reply:
Good point. Probably want to set the axes limits a bit differently I suppose... I guess maybe min(abs(myBounds, userVals)).

Reply:
@SeanPluemer This is the comment that needs to be addressed. Can you pull this PR and look into drawing the lines based on an arbitrary scale. My advice would be to use the current plot size as the default/ minimum and only adjust the axis if the data falls outside the range of the current x and y  limits as I show in my example screenshot above.